six/modules/c++/six/source/NITFWriteControl.cpp

```diff
@@ -98,12 +98,11 @@ NITFWriteControl::NITFWriteControl()
     loadXmlDataContentHandler();
 }
 
-void NITFWriteControl::initialize(Container* container)
+void NITFWriteControl::initialize(mem::SharedPtr<Container> container)
 {
     // Clean up
-    // NOTE  We do not own the container, so we don't delete 'mContainer' here
     mInfos.clear();
-
+    mContainer.reset();
```

Comment:
This line isn't needed... the assignment on the next line will take care of everything